src/query/query.rs

```diff
@@ -12,7 +13,16 @@ use crate::{DocAddress, Term};
 #[derive(Copy, Clone)]
 pub enum EnableScoring<'a> {
     /// Pass this to enable scoring.
-    Enabled(&'a Searcher),
+    Enabled {
+        /// The searcher to use during scoring.
+        searcher: &'a Searcher,
+
+        /// A [Bm25StatisticsProvider] used to compute BM25 scores.
+        ///
+        /// Normally this should be the [Searcher], but you can specify a custom
+        /// one to adjust the statistics.
+        statistics_provider: &'a dyn Bm25StatisticsProvider,
```

Comment:
Are there any measurements of whether this additional dynamic dispatch affects performance of the scoring?

Reply:
(No need to measure.)

Reply:
> This is being accessed at most per term, I think static dispatch will only matter for per doc stuff like collection.

Indeed that is a convincing argument that the additional dispatch will not matter. Thank you for taking the time to write it down!